Source/Additions/GSObjCRuntime.m

```diff
@@ -1334,7 +1334,7 @@ unsigned long long	(*imp)(id, SEL) =
                   }
                 val = [NSValue valueWithPoint: v];
               }
-            else if (GSSelectorTypesMatch(@encode(NSRange), type))
+            else if (strncmp(@encode(NSRange), type, strlen(@encode(NSRange))) == 0)
```

Comment:
I am not sure about the other types but at least for NSRange we need to make sure that CFRange may be used interchangeable. This really is a rare case but we should not break things.

Reply:
That makes sense. But there is also CGSize, CGRect, and CGPoint. The current approach is also not great and may lead to weird side-effects.

Reply:
What about skipping the first two characters (NS, CF, CG) and just compare the remaining string?

Reply:
A bit ugly but it will work

Reply:
You mean test that the encode type starts with '{' and leave out the next two characters? Sounds good enough for me.

Reply:
It turns out that CFRange and NSRange are not toll-free bridged on macOS. For instance, you cannot pass a CFRange to +[NSValue valueWithRange:]. Additionally, all geometry structs in Foundation are actually CoreGraphics structs. We should use the same naming scheme to support interop.

```c
#import <Foundation/Foundation.h>
#import <CoreFoundation/CoreFoundation.h>
#import <CoreGraphics/CoreGraphics.h>

int main() {
	@autoreleasepool {
		NSLog(@"%s", @encode(NSRect));
		NSLog(@"%s", @encode(CGRect));
		NSLog(@"%s", @encode(NSPoint));
		NSLog(@"%s", @encode(CGPoint));
		NSLog(@"%s", @encode(NSSize));
		NSLog(@"%s", @encode(CGSize));
		NSLog(@"%s", @encode(NSRange));
		NSLog(@"%s", @encode(CFRange));

		CGRect cgRect = CGRectMake(0, 1, 1, 1);
		CFRange range = CFRangeMake(0, 1);
		NSValue *val = [NSValue valueWithRange: range];
		NSLog(@"%@", val);
	}
	return 0;
}
```

```sh
hmelder@Hugos-MacBook-Pro ~ % clang test.m -framework Foundation -framework CoreFoundation -framework CoreGraphics
test.m:18:43: error: sending 'CFRange' to parameter of incompatible type 'NSRange' (aka 'struct _NSRange')
                NSValue *val = [NSValue valueWithRange: range];
                                                        ^~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSRange.h:45:38: note: passing argument to parameter 'range' here
+ (NSValue *)valueWithRange:(NSRange)range;
                                     ^
1 error generated.
```

And otherwise

```sh
2024-09-10 13:11:46.478 a.out[54713:2550694] {CGRect={CGPoint=dd}{CGSize=dd}}
2024-09-10 13:11:46.479 a.out[54713:2550694] {CGRect={CGPoint=dd}{CGSize=dd}}
2024-09-10 13:11:46.479 a.out[54713:2550694] {CGPoint=dd}
2024-09-10 13:11:46.479 a.out[54713:2550694] {CGPoint=dd}
2024-09-10 13:11:46.479 a.out[54713:2550694] {CGSize=dd}
2024-09-10 13:11:46.479 a.out[54713:2550694] {CGSize=dd}
2024-09-10 13:11:46.479 a.out[54713:2550694] {_NSRange=QQ}
2024-09-10 13:11:46.479 a.out[54713:2550694] {?=qq}
```

I've renamed all structs in NSGeometry.h accordingly.